// yun/src/webpack.config.js
/**
 * @author human.hjh
 * @created 15/9/14
 */
var path = require('path');
var webpack = require('webpack');
var currentUserPath = process.cwd();
var package = require(currentUserPath + '/package.json');


var entry = package['yun'] && package['yun']['entry'] || './example/index.js';

console.log(entry)

module.exports = {
	devtool: 'source-map',
	entry: [
		'webpack-dev-server/client?http://localhost:3000',
		'webpack/hot/only-dev-server',
			entry
	],

	plugins: [
		new webpack.HotModuleReplacementPlugin()
	],
	output: {
		path: path.join(currentUserPath, 'build'),
		filename: "index.js",
		publicPath: '/static/'
	},
	module: {
		loaders: [
			{
				test: /\.css$/,
				loader: ['style', 'css'],
				exclude: /node_modules/
			},
			{
				test: /\.js[x]?$/,
				//loaders: ['react-hot-loader', 'babel', 'jsx-loader?harmony'],
				loaders: ['react-hot-loader', 'babel'],
				exclude: /node_modules/
				//include: path.join(__dirname, 'src')
			},
			{
				test: /\.(png|jpg)$/,
				loader: 'url-loader?limit=8192'
			}
		]
	},
	resolve: {
		extensions: ['', '.js', '.jsx', '.json'],
		alias: {}
	}
};
